rotkehlchen/tests/unit/decoders/test_curve_lend.py

```diff
@@ -0,0 +1,694 @@
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.PAYBACK_DEBT,
+            asset=underlying_token,
+            balance=Balance(amount=FVal(repay_amount)),
+            location_label=user_address,
+            notes=f'Repay {repay_amount} crvUSD to Curve',
+            counterparty=CPT_CURVE,
+            address=string_to_evm_address('0xB5c6082d3307088C98dA8D79991501E113e6365d'),
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=11,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.APPROVE,
+            asset=underlying_token,
+            balance=Balance(amount=FVal(approve_amount)),
+            location_label=user_address,
+            notes=f'Set crvUSD spending approval of {user_address} by 0xB5c6082d3307088C98dA8D79991501E113e6365d to {approve_amount}',  # noqa: E501
+            address=string_to_evm_address('0xB5c6082d3307088C98dA8D79991501E113e6365d'),
+        ),
+    ]
+
+
+@pytest.mark.vcr(filter_query_parameters=['apikey'])
+@pytest.mark.parametrize('arbitrum_one_accounts', [['0x706A70067BE19BdadBea3600Db0626859Ff25D74']])
+def test_close_loan_using_collateral(
+        arbitrum_one_inquirer: 'ArbitrumOneInquirer',
+        arbitrum_one_accounts: list['ChecksumEvmAddress'],
+) -> None:
+    tx_hash = deserialize_evm_tx_hash('0xcca62e6832e2df6f4ff3ca5240069e2d4548dfd23ed8f08c448e4917b916d3df')  # noqa: E501
+    _, underlying_token = _create_arbitrum_vault_tokens(database=arbitrum_one_inquirer.database)
+    events, _ = get_decoded_events_of_transaction(evm_inquirer=arbitrum_one_inquirer, tx_hash=tx_hash)  # noqa: E501
+    timestamp, user_address, gas_amount, receive1_amount, receive2_amount = TimestampMS(1732139413000), arbitrum_one_accounts[0], '0.00000751744', '0.311352637896780974', '0.009899999999999997'  # noqa: E501
+    assert events == [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal(gas_amount)),
+            location_label=user_address,
+            notes=f'Burn {gas_amount} ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=13,
+            timestamp=timestamp,
+            location=Location.ARBITRUM_ONE,
+            event_type=HistoryEventType.RECEIVE,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=underlying_token,
+            balance=Balance(amount=FVal(receive1_amount)),
+            location_label=user_address,
+            notes=f'Receive {receive1_amount} crvUSD from Curve after repaying loan',
```

Comment:
Question: The closing/repaying does not seem to happen here. Where did it happen?

Reply:
The amount owed was repayed from the collateral already deposited in the contract. In this particular case, its a transaction I made while trying to figure out how it worked, and I had done a partial repayment in an earlier transaction. The transactions involved looked something like this:

1. borrowed 20 crvUSD using 0.01 WETH as collateral
2. repayed 20 crvUSD, but this didn't close the loan because I now also owed a small amount of interest
3. (the transaction in question here) closed the loan using the collateral deposited in the contract.